Classes/Emogrifier.php

```diff
@@ -296,12 +260,59 @@ public function emogrify()
             $this->removeInvisibleNodes($xpath);
         }
 
-        $this->copyCssWithMediaToStyleNode($cssParts, $xmlDocument);
+        $this->copyCssWithMediaToStyleNode($cssParts['media'], $xmlDocument);
 
         return $xmlDocument->saveHTML();
     }
 
     /**
+     * Parse a list of selectors from a string of css.
+     *
+     * @param string[] $css a string of raw CSS code
+     *
+     * @return array
+     */
+    private function parseSelectors($css)
+    {
+        $cssKey = md5($css);
+        if (!isset($this->caches[self::CACHE_KEY_CSS][$cssKey])) {
```

Comment:
This condition has a quite long content. Please let's change it to a guard clause.